blob/service.go

```diff
@@ -86,6 +93,67 @@ func DefaultSubmitOptions() *SubmitOptions {
 	}
 }
 
+func (s *Service) Start(context.Context) error {
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+	return nil
+}
+
+func (s *Service) Stop(context.Context) error {
+	s.cancel()
+	return nil
+}
+
+type SubscriptionResponse struct {
+	Blobs  []*Blob
+	Height uint64
+}
+
+func (s *Service) Subscribe(ctx context.Context, ns share.Namespace) (<-chan *SubscriptionResponse, error) {
+	if s.ctx != nil {
+		return nil, fmt.Errorf("service has not been started")
+	}
+
+	headerCh, err := s.headerSub(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	blobCh := make(chan *SubscriptionResponse, 3)
+	go func() {
+		defer close(blobCh)
+
+		for {
+			select {
+			case header, ok := <-headerCh:
+				if !ok {
+					log.Errorw("header channel closed for subscription", "namespace", ns.ID())
+					return
+				}
```

Comment:
headerSub might be closed because context is canceled, which is valid closure of sub and don't need to be logged with ERROR.
Moving handling of ctx errors higher than `!ok`  case should fix that.

Reply:
true, nice

Reply:
can you verify that the latest commit is what you had in mind?